### PR TITLE
i#4318 xarch memtrace: Add library search dir

### PR DESCRIPTION
**Pull request by derekbruening**
Adds a new option -alt_module_dir which specifies an alternate
directory in which to look for libraries in a modules.log offline
drcachesim file.  Hooks the option up to the opcode_mix and view tools
and to raw2trace (both its standalone launcher and drcachesim frontend
integration).

This will be useful in general, but is needed for adding tests with
checked-in raw files and binaries, where the checked-in modules.log
has a hardcoded path that will never match all test environments.

Adds a test that runs both raw2trace and opcode_mix on a new
checked-in aarch64 raw trace (this test will evolve for future
cross-arch testing, which is the reason for using aarch64).  The
binaries are stripped, with the two libraries "fake" binaries:
libmemtrace.so is not executed, and libdynamorio.so is only used for
dynamorio_sigreturn, so these are synthetic binaries (created from asm
files) to keep them smaller.  The tests deliberately fail on x86 and
match an error message that will improve when architecture tags are
added.

Issue: #4318

### CODE REVIEW

**Comment by derekbruening** [comment]
The modules.log is the wrong format for Windows (need extra columns for checksum, etc. there).  I'm just going to make the test UNIX-only.

File: clients/drcachesim/tracer/raw2trace.h [hunk 1]
```diff
@@ -439,6 +441,7 @@ class module_mapper_t final {
     byte *last_map_base_ = nullptr;
 
     uint verbosity_ = 0;
+    std::string alt_dir_;
```
Comment (snehasish) [inline clients/drcachesim/tracer/raw2trace.h:444]: s/alt_dir_/alt_module_dir_ to be consistent with the rest of the changes.

File: clients/drcachesim/common/options.cpp [hunk 2]
```diff
@@ -94,6 +94,12 @@ droption_t<std::string> op_module_file(
     "If the file is named modules.log and is in the same directory as the trace file, "
     "or a raw/ subdirectory below the trace file, this parameter can be omitted.");
 
+droption_t<std::string> op_alt_mod_dir(
```
[the remark this answers is not in the document]
Comment (derekbruening) [reply clients/drcachesim/common/options.cpp:97]: SGTM.  Done.

File: clients/drcachesim/tracer/raw2trace.h (hunk 1 above)
Comment (derekbruening) [reply clients/drcachesim/tracer/raw2trace.h:444]: Done.

**Pull request merged**